### DISCUSSION TITLE
Input from dbt Labs's core adapters team

### DISCUSSION THREAD

**Discussion by dataders** [databricks/dbt-databricks#203] [General]
as part of our new adapter verification process, the dbt core adapters engineering team and I met today to do a high-level review of the `dbt-databricks` codebase.

good news, your adapter passes the sniff test! That means that while the engineers haven’t yet had the time to review every single line of code (though they plan to in the next few sprints), we did not see anything that should stop the verification process.

An interesting output is that the team found opportunities where they might be better able to support, both of which are not of an urgent priority to fix:
1. the code introduced in #98 was difficult to grok at first (TIL  . is this to support both UC (that have database/catalog) and no-UC dbt-databricks (that do not have database/catalog) projects? I imagine that conditionality can pretty tricky, and the core team wanted to meet with y’all to discuss potential simpler solutions
2. this parsing of our exception message (see below) raised our hackles, in that now we know that things will break if we decide to standardize our Exception handling messages as we have already done with our logging. That said, we recognize the need for it, but wanted to tell you that we do plan to overhaul our Exception handling so that you won't even need this logic moving forward. stay tuned.
https://github.com/databricks/dbt-databricks/blob/dbd58fb4e15006908a9b6f152b1334adc73c89d6/dbt/adapters/databricks/impl.py#L114-L125

lastly, are there any particular areas you’d like us to review more closely? Are there any ways that the dbt-core adapters team can better support dbt-databricks now and in the future? let us know!

**Comment by gatorsmile** [comment]
@dataders Thank you! 

What is the new adapter verification process? Could you provide more info about the process?

**Reply by dataders** [reply]
So here's what we have public-facing right now [Verifying a new adapter](https://docs.getdbt.com/docs/contributing/adapter-development/7-verifying-a-new-adapter). Reach out to @superdupershant and he can give you more info that we've shared with him.

**Comment by dataders** [comment]
We also noticed that the integration test suite from dbt-core is defined, but not being run as part of CI (see #204). If there was one thing to fix before the dbt-databricks is officially verified, it is this. 👀  @gatorsmile @ueshin @superdupershant

**Comment by ueshin** [comment]
> 1. the code introduced in #98 was difficult to grok at first (TIL  . is this to support both UC (that have database/catalog) and no-UC dbt-databricks (that do not have database/catalog) projects? I imagine that conditionality can pretty tricky, and the core team wanted to meet with y’all to discuss potential simpler solutions

Yes, `jinja2.runtime.Undefined` was troublesome when supporting both UC and non-UC. For example, we needed to make the include policy for `database` be `True`, but if it's `Undefined`, the rendered string will be `.schema.table` and cause to fail the query. #98 tries to replace `Undefined` with `None` when the macro calls `@available` adapter methods.

It would be great if `dbt-core` could handle or provide the way to handle `Undefined` whenever macros call Python functions/methods.
